app/controllers/student_labs_controller.rb

```diff
@@ -15,18 +17,20 @@ def create
       redirect_to student_labs_path
       return
     end
+    # puts "REQUEST: #{request.remote_ip}"
     @student_lab = current_user.student_labs.find_by(:lab_id=>student_lab_params[:lab_id])
     if @student_lab.present?
       @student_lab.update(student_lab_params.merge(complete: true))
     else
       @student_lab = current_user.student_labs.create(student_lab_params.merge(complete: true))
     end
-    flash[:success] = "Submission success"
+    @student_lab.update(:ip=>request.remote_ip)
```

Comment:
This can be combined into the line 25, do `@student_lab = current_user.student_labs.create(student_lab_params.merge(complete: true, ip: request.remote_ip))`